build-system/common/esbuild-babel.js

```diff
@@ -48,57 +99,44 @@ function getEsbuildBabelPlugin(
   preSetup = () => {},
   postLoad = () => {}
 ) {
-  function sha256(contents) {
+  function sha256(contents, options) {
     if (!enableCache) {
       return '';
     }
     const hash = crypto.createHash('sha256');
-    hash.update(callerName);
+    hash.update(JSON.stringify(options));
     hash.update(contents);
     return hash.digest('hex');
   }
 
   function batchedRead(path) {
     let read = readCache.get(path);
     if (!read) {
-      read = fs.promises
-        .readFile(path)
-        .then((contents) => {
-          return {
-            contents,
-            hash: sha256(contents),
```

Comment:
Doing a single file hash here was important, these files are large. Now each reader in the batch will do their own.

Reply:
When does the `batch` really get used? From testing it doesn't get used during the `build` path

Reply:
It was important for test suites. It happens when there are multiple entry points, each entry point is running the full compile chain for every file (so, N compiles of the same file if it's used by each entry point's graph)

Reply:
I've optimized the hashing s.t. it only takes ~20ms now.